docs/content/sips/013-sqlite.md

```diff
@@ -74,13 +76,17 @@ query: func(conn: connection, query: string, parameters: list<value>) -> expecte
 // Close the specified `connection`.
 close: func(conn: connection)
 
-// A result of a query
-record query-result {
-  // The names of the columns retrieved in the query
-  columns: list<string>,
-  // The row results each containing the values for all the columns for a given row
-  rows: list<row-result>,
-}
+// A result of making a query
+type query-result = u32
+
+// Get columns for a given `query-result`
+get-columns: func(query-result: query-result) -> expected<list<string>, error>
+
+// Get a `row-result` at a given index for a given `query-result`
+get-row-result: func(query-result: query-result, index: u32) -> expected<option<row-result>, error>
```

Comment:
Do sqlite results support random access? I thought they had cursor/iterator semantics?

Reply:
The implementation does not mirror the underlying semantics of sqlite - we just keep things in memory on the host side which makes random access possible. We could do more closely mirror the sqlite API if we want though...

Reply:
+1 for cursor/iterator if we go down the one row at a time route.